### PR TITLE
Updated to work with XR plugin management (non-legacy) 

### PR DESCRIPTION
**Pull request by roblkw-asu**
Edited the WebXRCamera.cs and WebXRController.cs scripts to work with modern XR plugin management (non-legacy).

Also updated the readme for In-Editor VR testing.

### CODE REVIEW

**Comment by De-Panther** [comment]
Thanks for the PR.
Which versions of Unity this change supports?

**Comment by roblkw-asu** [comment]
2019.1 and later.

**Comment by De-Panther** [comment]
@fernandojsg maybe it's time to drop the support for 2018.4?

But even then, how many devs use the old APIs rather than the new one?

A solution could be:
- Remove the current `WebXRCamera` and `WebXRController`. Let the dev get the Pose of camera/controllers, and let him decide how to implement them in his project.
- After removing the current classes, adding our own samples of implementing them, that works with both the old and new APIs. We can use Unity's Assembly Definitions files to determine which API is currently active, and select the relevant API to use.
- Adding Assembly Definitions files to this PR and use them to select which API to use.

What does the both of you think?

**Comment by roblkw-asu** [comment]
Those are good ideas. However, I believe that Unity's 2019.x APIs support
the legacy XR use as well without any changes.

I loaded my project *without* adding the modern Unity XR Plugins, and it
worked just fine on my Oculus Quest (on Link). Of course, it does then
still need the deprecated Project Settings checked.

(By the way, thanks for the excellent hooks into WebXR. I think this is an
exciting direction for releasing VR apps.)

--
Robert LiKamWa
Assistant Professor, Arizona State University
School of Arts, Media and Engineering
School of Electrical, Computer and Energy Engineering
http://meteor.ame.asu.edu

**Comment by fernandojsg** [comment]
I'm ok with deprecating support for the older version and both the legacy XR API. 
WebXR is a fairly new API and it's continually improving so I believe they should understand that it's ok to be updated to the latest browser as well as to the latest Unity.
Also due to the scope of this project, it should be easier to maintain with less backwards compatibility, adding support for something that is going to be deprecated soon enough it's something I don't think we should use our limited time for.
So +1 to the proposal by @De-Panther

**Comment by StephenHodgson** [comment]
+1 on the assembly definitions

**Comment by De-Panther** [comment]
Ok, so lets start with this PR. which use the new API, and if someone will want to add support for the older API using Assembly Definitions, they are more than welcome.

File: Assets/WebXR/Scripts/WebXRCamera.cs [hunk 1]
```diff
@@ -31,11 +33,11 @@ void OnEnable()
...
-#if UNITY_EDITOR
-			// No editor specific funtionality
-#elif UNITY_WEBGL
-			postRenderCoroutine = StartCoroutine(endOfFrame());
-#endif
+            #if UNITY_EDITOR
+			            // No editor specific funtionality
+            #elif UNITY_WEBGL
+			            postRenderCoroutine = StartCoroutine(endOfFrame());
+            #endif
```
Comment (De-Panther) [inline Assets/WebXR/Scripts/WebXRCamera.cs:40]: Please revert those indentation changes

File: Assets/WebXR/Scripts/WebXRCamera.cs [hunk 2]
```diff
@@ -79,5 +81,34 @@ private void onVRChange(WebXRState state)
                 cameraR.projectionMatrix = rightProjectionMatrix;
             }
         }
+
+        //Update Camera position according to Unity XR, if not using WebGL
+        private void Update()
```
Comment (De-Panther) [inline Assets/WebXR/Scripts/WebXRCamera.cs:86]: This should work only in Editor? Or also in other platforms?
Please add
`#if UNITY_EDITOR`
or
`#if UNITY_EDITOR || !UNITY_WEBGL`

File: Assets/WebXR/Scripts/WebXRController.cs [hunk 3]
```diff
@@ -268,31 +268,85 @@ private void SetVisible(bool visible)
...
+        void InitXR()
+        {
+            xr_inited = true;
+            List<XRInputSubsystem> subsystems = new List<XRInputSubsystem>();
+            SubsystemManager.GetInstances<XRInputSubsystem>(subsystems);
+            for (int i = 0; i < subsystems.Count; i++)
+            {
+                subsystems[i].TrySetTrackingOriginMode(TrackingOriginModeFlags.Floor);
+            }
+        }
```
Comment (De-Panther) [inline Assets/WebXR/Scripts/WebXRController.cs:284]: Add empty line after

File: Assets/WebXR/Scripts/WebXRController.cs [hunk 4]
```diff
@@ -268,31 +268,85 @@ private void SetVisible(bool visible)
...
         void Update()
         {
             // Use Unity XR Input when enabled. When using WebXR, updates are performed onControllerUpdate.
-            if (!XRDevice.isPresent) return;
-
```
Comment (De-Panther) [inline Assets/WebXR/Scripts/WebXRController.cs:275]: Maybe we should also add
`#if UNITY_EDITOR`
or
`#if UNITY_EDITOR || !UNITY_WEBGL`
Before this `Update()` call

File: Assets/WebXR/Scripts/WebXRCamera.cs [hunk 5]
```diff
@@ -79,5 +81,34 @@ private void onVRChange(WebXRState state)
...
+                    }
+                }
+                cameraMain.transform.localPosition = mHeadPos;
+                cameraMain.transform.localRotation = mHeadRot.normalized;
+            }
+        }
     }
-}
\ No newline at end of file
+#endif
```
Comment (De-Panther) [inline Assets/WebXR/Scripts/WebXRCamera.cs:113]: #endif should be after the closing bracket of `Update()` and not the closing bracket of `WebXRCamera` class

**Pull request merged**